packages/core-components/src/layout/Content/Content.tsx

```diff
@@ -55,9 +56,14 @@ type Props = {
 
 export function Content(props: PropsWithChildren<Props>) {
```

Comment:
The `Content` component isn't really guaranteed to exist, and I also think it's semantically quite far away from the `Sidebar`. I think my favorite solution to this problem would be something that only adds to the "sidebar world", meaning that if you don't use the sidebar, then none of the logic to handle this remains.

We do already have the `SidebarPage`, and it feels like that's the place where we want to add any context providers. Maybe we can just end up with something that looks like this?

```tsx
<SidebarPage>
  <Sidebar>
    ...
  </Sidebar>
  <SidebarPage.Content>
    {children}
  </SidebarPage.Content>
</SidebarPage>
```

Migration would be somewhat smooth I think as we'd just make the `SidebarPage.Content` optional and not support the content focus if it's not there. I think perhaps the most tricky bit is to make sure that doesn't have a big impact layout-wise since we'd be adding in an extra element.

Reply:
I understand that the `Content` component could not exist, if not I don't see any problem the `focusContent` would simply make no actions

I've tried the SidebarPageContent approach here #7232 but there are 2 things that I like more from this approach:

- SidebarPageContent wraps the whole plugin page, and when tab will move to the first focus area but depending on each plugin design this could be nice or not, the focus could end up on some aside notifications instead of the real content. Having this in the `Content` component each plugin could decide where is the content directly
- We are using this `focusContent` from the Sidebar on these use cases, but the content ref has nothing to do with the Sidebar, maybe is more a naming thing


we could still use the `SidebarPage` but we would need to add a different context or a mixup with `SidebarPinStateContext`

Reply:
If using the `Content` in order to declare where the real content makes sense. I can remove the `LayoutProvider`, a and move that context inside `SidebarPage` component:


```diff
+ const ContentContext = createContext<>({
+    contentRef: undefined,
+});

export function SidebarPage(props: PropsWithChildren<{}>) {
  // ....
  return (
    <SidebarPinStateContext.Provider
      value={{
        isPinned,
        toggleSidebarPinState,
      }}
    >
+       <ContentContext.Provider value={{ content }}>
             <div className={classes.root}>{props.children}</div>
+      </ContentContext.Provider>
    </SidebarPinStateContext.Provider>
  );
}

```

Do you think that we could still consider renaming the SidebarPage in the future? not sure if "Layout" is a good name, but couldn't come with any better

Reply:
Yep, makes sense to use `Content` for now and have it be an optional thing. Let's move the provider into the `SidebarPage` as you suggested, and yeah I think it could use a rename at some point :grin:

We will end up with duplication issues for `core-components` that breaks the context, which we already see happening every now and then with the sidebar. I think we really want to move over the `SidebarContext` to be a more core part of the package and give it a `version-bridge`, but it's gonna be a breaking change that I think we should do later.